wallet/src/repository/wallet/mod.rs

```diff
@@ -1659,9 +1707,16 @@ where
             _ => 0,
         };
 
+        // Insert resolved inputs into transaction metadata so that they can be added to the balance
+        // movement.
+        let txn = model::ExtendedTransaction {
+            metadata: Some(model::TransactionMetadata::InputValues(resolved_inputs)),
+            transaction: txn.transaction.clone(),
+        };
```

Comment:
This looks like the most probable cause for the `WrongMetadataType` error, the metadata is always set to `InputValues` here. Since tally transactions have no inputs, this conversion shouldn't be needed if the transaction is a tally transaction.

Reply:
Alright, this seems to fix it:

```rust
// Insert resolved inputs into transaction metadata so that they can be added to the balance
// movement.
let txn = match &txn.metadata {
    Some(model::TransactionMetadata::InputValues(_input_values)) => {
        ExtendedTransaction {
            metadata: Some(model::TransactionMetadata::InputValues(resolved_inputs)),
            transaction: txn.transaction.clone(),
        }
    }
    _ => {
        txn.clone()
    }
};
```

The `txn.clone()` could be avoided but it makes the code a bit uglier.

Reply:
Nice catch. I modified it so it never overwrites metadata if there's some in place (which is always the case for tally transactions).